pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptions.java

```diff
@@ -82,4 +98,24 @@ public static Long getTimeoutMs(Map<String, String> queryOptions) {
       return null;
     }
   }
+
+  @Nullable
+  public static Boolean getEnableSegmentTrim(Map<String, String> queryOptions) {
+    String enableSegmentTrim = queryOptions.get(Request.QueryOptionKey.SEGMENT_ENABLE_TRIM);
+    if (enableSegmentTrim != null) {
+      return Boolean.parseBoolean(enableSegmentTrim);
+    } else {
+      return null;
+    }
+  }
```

Comment:
```suggestion
  public static boolean getEnableSegmentTrim(Map<String, String> queryOptions) {
    return Boolean.parseBoolean(queryOptions.get(Request.QueryOptionKey.SEGMENT_ENABLE_TRIM));
  }
```

Reply:
Has to bring back null check to avoid nullpointerexception

Reply:
While that's not the case. `Boolean.parseBoolean()` will never through NPE